HelpSource/Classes/Object.schelp

```diff
@@ -200,6 +200,8 @@ A String containing the resulting file's path.
 
 subsection::Equality and Identity
 
+Equality, queried with code::==:: or code::!=::, is whether the values or contents of two objects are the same by comparing the values, contents or emphasis::hash table indecise::. Identity, queried by code::===:: or code::!==::, is whether two objects refer to the same object by comparing the emphasis::identity hash table indecise:: or virtual memory addresses.
+
```

Comment:
Thank you for this. 

- I arranged your sentences and example codes a bit.
- also used the sentences by @JordanHendersonMusic !

Reply:
> Equality, tested for with code::==:: or code::!=:: is a relation between objects. So code::a == b:: is true when the object in the variable `a` and the object in `b` are equivalent in some essential aspect. Two strings, for example, are equal when they have the same characters in the same order. Identity, by contrast, is a relation that every object has to itself. So code::a === b:: is true only when the the variables `a` and `b` both contain the same object.
> 
> code:: // symbols, written with single quotes, are special: there is only one of each // therefore, no two symbols are equal, but they are always different "we are all the same" == "we are all the same" // true "we are all the same" === "we are all the same" // false: same, but not identical // two representations of the same symbol are identical 'we are all the same' === 'we are all the same' // true ::

That's something math and computation not always say the same thing.
For example, you described (sort of) associativity and identity morphism. Every object is idential to itself, and that's (in this sense, in SC, that is true up to the point they are the same data on the same memory address). But there is no other concept of identical morphism, which means a different thing, I *think*... 
]

Reply:
Ok, fair, but those 1 are not even assigned to anything, which would break this identity. If you just evaluate an (not assigned) integer twice, this should be be the result one expect. 

to be clear, I'm not defending introducing "memory"  as a topic here.

Reply:
> but those 1 are not even assigned to anything, which would break this identity.

This isn't true either.
```
a = 1
b = 1
a === b // true
```
Identity is define like `(a->tag == b->tag) && (a->u.i == b->u.i);`  meaning, if the tags are the same, and the raw byte representation is the same, then they are the identical.

Reply:
fair enough, thanks. so identity is also characterized by a 'fixed point' of a functor could also be "similar enough" to identity. 

example: applying f(x) to an object x returns an object that is effectively the same as x (perhaps up to some transformation that is reversible), x could be considered a fixed point of f(x). This could be seen in cases where an operation doesn't change the essential nature or identity of an object. 

Identity morphisms would also be expanded to identity operations, why not?

Reply:
For comparison

```

-- everything is identical here:
main :: IO ()
main = do
  print (id (addOne 5))      -- Output: 6
  print ((id . addOne) 5)    -- Output: 6
  print ((addOne . id) 5)    -- Output: 6
  
  
-- this library is not testing if things are the same 'object'
-- it is testing precisely the memory location, nothing more

data Somebody = Somebody { identifier :: String, age :: Int } deriving (Eq, Show)

main :: IO ()
main = do
    let entity1 = Somebody "Zephyr" 30
    let entity2 = entity1
    let entity3 = Somebody "Zephyr" 30

    sn1 <- makeStableName entity1
    sn2 <- makeStableName entity2
    sn3 <- makeStableName entity3

    print (sn1 == sn2)  -- True
    print (sn1 == sn3)  -- False

```